Fix Strtab::insert panic

josephlr opened the pull request:
Right now  `Strtab::insert` will panic with and out-of-bounds slice if Strtab.data starts with a null byte, and Strtab::insert is called with a non-nil value.

This change fixes this bug in `Strtab::insert` by cleaning up the code, and removing an unnecessary vector. If you want me to just remove the hashtable logic entirely (as per the TODO) just let me know.

It also does a fixes a few warnings and adds a `.gitignore`, if you want me to split out these changes into a separate PR, just let me know.

josephlr committed:
Add .gitignore

josephlr committed:
Fix `unused_parens` warnings

josephlr committed:
Improve Strtab::insert

This code was crashing if Strtab::data began with a null byte

aep commented:
thank you for the PR!
i haven't touched this in a while, so it would greatly help if you could add a test that is fixed with this PR

josephlr commented:
> thank you for the PR!
> i haven't touched this in a while, so it would greatly help if you could add a test that is fixed with this PR

I added the tests and also removed the hashtable logic (per the TODO). There were also some minor cleanups I did. I also ran rustfmt on the file.

aep reviewed (approved):
Thank you!
